cpp/src/parquet/encoding.cc

```diff
@@ -3988,4 +3988,21 @@ std::unique_ptr<Decoder> MakeDictDecoder(Type::type type_num,
 }
 
 }  // namespace detail
+
+// Informed heavily by https://github.com/apache/parquet-format/blob/master/Encodings.md
+// Should we also consider if we're in dictionary encoding mode? or assume no for the
+// moment?
+Encoding::type ChooseFallbackEncoding(Type::type data_type,
+                                      ParquetVersion::type parquet_version,
+                                      ParquetDataPageVersion datapage_version) {
+  if (data_type == Type::BOOLEAN && parquet_version != ParquetVersion::PARQUET_1_0 &&
+      datapage_version == ParquetDataPageVersion::V2) {
+    return Encoding::RLE;
+  } else if (data_type == Type::BYTE_ARRAY) {
```

Comment:
Do you want to support other data types as well?

Reply:
I do, though it's not immediately clear to me what the best encoding to choose for other data types is. The docs make it crystal clear in the case of `BYTE_ARRAY`, but not necessarily other types as far as I can tell.

Reply:
Should we only goes to this in v2?

Reply:
@ClifHouck What about following what parquet-mr does? Although parquet-mr has bundled data page version and writer version, we can either do the following:
- Enable v2 encodings if writer version is 2.0+
- Enable v2 encodings when writer version is 2.0+ and data page version is v2.